sling.h

```diff
@@ -26,6 +26,8 @@ THE SOFTWARE.
 #include <functional>
 #include <map>
 
+#include "slingdefs.h"
```

Comment:
I think it's best that we keep the library a single header one for ease of use, so maybe add the new defines in `sling.h` directly?